### ISSUE TITLE
Ember 2.10.0 - Route's redirect method issue. Incorrect arguments. BUG

### ISSUE THREAD

**Issue by Exelord** [emberjs/ember.js#14660]
Hi,
According to the doc: http://emberjs.com/api/classes/Ember.Route.html#method_redirect

The `redirect()` method should has an args: `model` and `transition`
But in fact, the `model` is serialized as a `string`.

Eg:
```hbs
{{link-to 'projects.project' project}}
```

```js
redirect(model) {
  console.log(model) // returns "<myapp@model:project::ember2186:182>"
}
```

In previous releases, it returned correct `object` model not a serialized one.
Is it bug or new feature? ;)

**Comment by GavinJoyce** [comment]
@Exelord I tried to recreate this in a twiddle, but didn't manage to. Perhaps you could try modify this twiddle to demonstrate the issue you're having?

https://ember-twiddle.com/d5c1ff175b24e49a46876e658b46d9f0?openFiles=templates.application.hbs%2C&route=%2Froute2%2F1

**Comment by Exelord** [comment]
https://ember-twiddle.com/d5c1ff175b24e49a46876e658b46d9f0?openFiles=routes.tasks.new.js%2C&route=%2Ftasks%2Fnew

is the `queryParams` correct results? Compared to 2.9 now they are serialized.

**Comment by GavinJoyce** [comment]
Are you referring to `&route=%2Froute1%2F1`? If so, that's embertwiddle's URL and it appears to be URL encoded correctly.

**Comment by GavinJoyce** [comment]
I just noticed that you have a route `tasks/new` in you twiddle url, but I'm not seeing those routes. There may be a bug in ember twiddle with saving, you may have to fork my twiddle and safe if that's the case

**Comment by pixelhandler** [comment]
@Exelord In your Twiddle link when I change to the release version I still see the object logged, not a string representation of the model.

**Comment by Exelord** [comment]
@GavinJoyce indeed. It was a problem with ember twiddle. My changes just gone. Tomorrow morning I will just write a new one.

**Comment by Exelord** [comment]
@GavinJoyce Ok, here it is.
https://ember-twiddle.com/075a9d54c5947e1fb64aaae7e405981a?openFiles=templates.application.hbs%2C&route=%2Ftasks%2F1

On 2.10 objects in query-params are serialized on 2.9 are not.
Probably it's a bug fix if I good understand but how should I pass now optional object to route?

**Comment by pixelhandler** [comment]
@Exelord @GavinJoyce is this still an issue, perhaps we should close or create a new reproduction of this, what do you think?
